torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1197,53 +1197,59 @@ def sample_inputs_linalg_vector_norm(op_info, device, dtype, requires_grad, **kw
 
     return inputs
 
-# In order to use the kwarg alpha, partials should be used in an OpInfo's sample_inputs_func
-# eg. sample_inputs_func=partial(sample_inputs_binary_pwise, alpha=2)
-# Then one sample input would also be generated corresponding to the value of alpha provided.
-# In the future, kwargs 'alpha_floating', 'alpha_integral' & 'alpha_complex' can be used to
-# specify scalars of floating, integral & complex types as values for "alpha".
-# Keyword argument `rhs_exclude_zero` is used to exclude zero values from rhs tensor argument
-# This is necessary for operations like `true_divide`, where divide by zero throws an exception.
-def sample_inputs_binary_pwise(op_info, device, dtype, requires_grad, extra_kwargs=None, **kwargs):
-    if extra_kwargs is None:
-        extra_kwargs = {}
-
-    scalar = 3.14 + 3.14j if dtype.is_complex else (3.14 if dtype.is_floating_point else 3)
-    scalar = 1 if dtype is torch.bool else scalar
-    tests_list = [
-        ((S, S, S), (S, S, S), False),
-        ((S, S, S), (S, S), False),
-        ((), (), False),
-        ((S, S, S), (), False),
-        ((S, S, S), scalar, False),
-        ((), scalar, False)
+
+# Metadata class for binary "universal functions (ufuncs)" that accept two
+# tensor and have common properties
+class BinaryUfuncInfo(OpInfo):
+    """Operator information for 'universal binary functions (binary ufuncs).'
+    These are functions of two tensors with common properties like:
+      - they are elementwise functions
+      - the output shape is determined by the input shape
+      - they typically have method and inplace variants
+      - they typically support the out kwarg
+      - they typically have NumPy or SciPy references
+    See NumPy's universal function documentation
+    (https://numpy.org/doc/stable/reference/ufuncs.html) for more details
+    about the concept of ufuncs.
+    """
+    pass
+
+
+def sample_inputs_binary_pwise(
+        op_info, device, dtype, requires_grad, op_kwargs=None, lhs_make_tensor_kwargs=None, **rhs_make_tensor_kwargs
```

Comment:
Model `lhs_make_tensor_kwargs` and `rhs_make_tensor_kwargs` symmetrically. It's kinda weird that `rhs_make_tensor_kwargs` captures everything left over.

Reply:
> It's kinda weird that `rhs_make_tensor_kwargs` captures everything left over.

Agreed. Still, if I'm not mistaken, the function needs to take a `**kwargs` argument. How would we handle that? Ignore it?

Reply:
It can still take **kwargs after lhs_make_tensor_kwargs and rhs_make_tensor_kwargs, right? But then yes this should ignore the kwargs dict.